### DISCUSSION TITLE
Mapping user data to body in onContactAdded

### DISCUSSION THREAD

**Discussion by LeXXik** [jrouwe/JoltPhysics.js#74] [Q&A] [answered]
In order to match the visual shape of a rendering framework with the physical body, I would generate and store an index. That index would allow me to map the body and find the correct entity to update later.

I would store the generated index at the shape level (because body doesn't allow storing user data?). Currently I do this to find the map index:
```
body.GetShape().GetUserData();
```

When I am handling `OnContactRemoved`, though, I am not able to get it, but it allows me to get a body ID. What would be a way to get my index from the `OnContactRemoved` closure? Does a physics system have a method of returning a body, by giving it an ID, something like `system.GetBodyByID(id);`?

**Comment by kutu** [comment, marked as answer]
maybe this https://github.com/jrouwe/JoltPhysics.js/discussions/18#discussioncomment-7224515

**Comment by LeXXik** [comment]
Thanks! That works for me.
